db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
i understand the case for expanding while 0 overlapping files.

expanding as long as 1 overlapping file seems somewhat arbitrary. maybe people configure a small L1 file size, then the hot L0 key-range might always overlap with, e.g., 3 L1 files, in which case we want the same optimization. to generally handle this case I think we need the logic that minimizes work-per-deleted-file.

so i'd be happy with any of (1) expanding inputs as long as 0 overlapping L1 files, (2) expanding inputs as long as work-per-deleted-file decreases, (3) expanding inputs aggressively and thus dropping the optimization. do any of these sound ok to you?